blueprints/src/main/java/dev/streamx/aem/connector/blueprints/FranklinCheck.java

```diff
@@ -0,0 +1,29 @@
+package dev.streamx.aem.connector.blueprints;
+
+import com.day.cq.wcm.api.Page;
+import org.apache.sling.api.resource.Resource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Optional;
+
+class FranklinCheck {
+
+  private static final String FRANKLIN_RES_TYPE_REGEX = "core/franklin/components/page/.+";
+  private static final Logger LOG = LoggerFactory.getLogger(FranklinCheck.class);
+
+  private final Page pageToCheck;
+
+  FranklinCheck(Page pageToCheck) {
+    this.pageToCheck = pageToCheck;
+  }
+
+  boolean isFranklinPage() {
+    boolean isFranklinPage = Optional.ofNullable(pageToCheck.getContentResource())
+        .map(Resource::getResourceType)
```

Comment:
Shouldn't it check resource type including hierarchy? Strings comparison won't work with inheritance.